implants/imix/src/task.rs

```diff
@@ -25,178 +17,47 @@ impl TaskHandle {
         TaskHandle {
             id,
             runtime,
-            download_handles: Vec::new(),
+            pool: tokio::task::JoinSet::new(),
         }
     }
 
     // Returns true if the task has been completed, false otherwise.
     pub fn is_finished(&self) -> bool {
-        // Check File Downloads
-        for handle in &self.download_handles {
-            if !handle.is_finished() {
-                return false;
-            }
+        // Check Report Pool
+        if !self.pool.is_empty() {
```

Comment:
If a process fails to send any output over a callback interval is it considered finished? Is this true if it doesn't send anything for n callback intervals?

Reply:
The pool is empty if there are no currently running tokio tasks for reporting output. If output fails to send, in theory the tokio task would exit (with an error), but that is the responsibility of the transport (e.g. retry logic).

Reply:
Okay I misunderstood I thought this pool was for channels not the actual tasks.